### PR TITLE
Added scheduler cloud service CRUD operations

### PR DESCRIPTION
**Pull request by antonydenyer**
#214 Added scheduler cloud service CRUD operations. Will implement the jobCollection functionality and wanted to check I was heading in the right direction and clear any MS hurdles.

### CODE REVIEW

File: azure/servicemanagement/schedulermanagementservice.py [hunk 1]
```diff
@@ -69,8 +75,71 @@ def list_cloud_services(self):
         return self._perform_get(self._get_list_cloud_services_path(),
                                  CloudServices)
 
+    def create_cloud_service(self, cloud_service_id, label, description, geo_region):
+        '''
+        The Get Cloud Service operation gets all the resources (job collections)
```
Comment (huguesv) [inline azure/servicemanagement/schedulermanagementservice.py]: Wrong description?

**Comment by huguesv** [comment]
Thanks for the PR!  Looks fine so far.  Let us know when it's ready.

**Comment by antonydenyer** [comment]
Having some trouble with the create job endpoint. It doesn't fit the current model as it only accepts json, had to put some switches in for content-type and some other bits. Currently thinking the signature will look something like:

```
    def create_job(self, cloud_service_id, job_collection_id, job_id, job):
        '''
        The Create Job request creates a new job.
        cloud_service_id:
            The cloud service id
        job_collection_id:
            Name of the hosted service.
        job_id:
            The job id you wish to create.
        job:
            A dictionary of the payload
        '''
```

What are your thoughts?

**Comment by huguesv** [comment]
Using a dict for the job parameter should save you a bunch of serialization work (compared to creating model classes).

We're working on a code generator and should have our first generated python clients in a few months (3-4 months?).  Scheduler should be one of them, but it's up to each service team to produce the metadata for the generator, so we don't know when that will ready.

My guess is that you need this code now.  If you continue the work and submit it, we'll take it, just understand that it will be obsoleted once we have the generated version of the scheduler api.

**Comment by antonydenyer** [comment]
@huguesv totally understand that it will be deprecated in a few months. Because of this I haven't spent any time refactoring and haven't implemented all the methods for the API, just the ones that I need. 

Note that the ```get_job``` returns a dict but doesn't do anything parsing, it just ```json.loads``` the body. And as suggested ```create_job``` takes a dict for the job payload.

Let me know if you want any other changes.

File: azure/servicemanagement/__init__.py [hunk 2]
```diff
@@ -1872,6 +1872,14 @@ def __init__(self):
         self.resources = Resources()
 
 
+class JSONEncoder(json.JSONEncoder):
```
Comment (huguesv) [inline azure/servicemanagement/__init__.py]: This probably should be marked as private/internal (name it _JSONEncoder)

**Comment by huguesv** [comment]
That's great!  Looks good. If you can figure out what's not working with `check_job_collection_name`, great, if not I can try when it's in.  If I can't get it working, I'll delete or comment out the function.  Can you confirm this works on both Python 2.7 and 3.3/3.4?

**Comment by antonydenyer** [comment]
It was developed and tested using python 2.7 on OS X Yosemite. I've tried running the tests under python 3.4.3 but I'm getting ``` from util import (
ImportError: No module named 'util' ```  
I created a virtualenv using ``` mkvirtualenv --python=/usr/local/bin/python3.4 azure-sdk ```
and I'm running ```nosetests``` from the root of the project.
```
which nosetests
~/.virtualenvs/azure-sdk/bin/nosetests
```

It's the first time I've done anything with python so it could be something really obvious that I haven't done. If there's any resources you could point me to that might be useful as well.

**Comment by huguesv** [comment]
No worries.  I'll try it out.  In PTVS, we set PYTHONPATH to include the tests folder, so that's probably why it works for me.  On Linux/Mac, I use the tests/run script, which I run from the context of the tests folder.  I'll merge this and adjust the code if needed.

**Pull request merged**